### PR TITLE
Improve tests for BigInt.prototype.valueOf

### PR DESCRIPTION
**Pull request by leobalter**

### CODE REVIEW

File: test/built-ins/BigInt/prototype/valueOf/return.js [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
 esid: sec-bigint.prototype.valueof
-description: BigInt.prototype.valueOf this value coercion
-info: >
+description: >
+  BigInt.prototype.valueOf returns the primitive BigInt value.
+info: |
   BigInt.prototype.valueOf ( )
 
   Return ? thisBigIntValue(this value). 
 includes: [typeCoercion.js]
```
Comment (leobalter) [inline test/built-ins/BigInt/prototype/valueOf/return.js]: I need to remove the includes... That's why the linter triggered an error

**Comment by leobalter** [comment]
@spectranaut the commits d729577 and b307b80 should fix the remaining parts.

File: test/built-ins/BigInt/prototype/valueOf/return.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  BigInt.prototype.valueOf ( )
+
+  Return ? thisBigIntValue(this value). 
+features: [BigInt]
+---*/
+
+var valueOf = BigInt.prototype.valueOf;
+
+assert.sameValue(valueOf.call(0n), 0n, "0n");
+assert.sameValue(valueOf.call(Object(0n)), 0n, "Object(0n)");
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/valueOf/return.js]: Thanks!

File: test/built-ins/BigInt/prototype/valueOf/this-value-invalid-primitive-throws.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}, "number (1)");
+
+assert.throws(TypeError, function() {
+  valueOf.call(NaN);
+}, "NaN");
+
+var s = Symbol();
+assert.throws(TypeError, function() {
+  valueOf.call(s);
+}, "symbol");
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/valueOf/this-value-invalid-primitive-throws.js]: Very clear, glad to see the helper going away where possible. Thanks for taking the extra time to refine these.

**Comment by littledan** [comment]
Note that although there was only one test checked in which used this infrastructure, there are more places in the spec where it would be applicable, namely in BigInt.prototype.toString and toLocaleString. cc @thejoshwolfe @cxielarko

**Comment by leobalter** [comment]
There is a PR for toString already which I'm working to apply fixes at this very moment. We should have an update soon.

**Pull request merged**